Sources/TuistKit/Generator/Generator.swift

```diff
@@ -28,6 +28,7 @@ class Generator: Generating {
     private let sideEffectDescriptorExecutor: SideEffectDescriptorExecuting
     private let configLoader: ConfigLoading
     private let manifestGraphLoader: ManifestGraphLoading
+    private var lintWarningIssues: [LintingWarningIssue] = []
```

Comment:
@tuist/core do you know if having a stateful generator may cause issues? Do we have cases where we run the generator multiple times in a row?

Reply:
> Do we have cases where we run the generator multiple times in a row?

I don't think we do but it would still be good if we could get rid of the state here (not sure if feasible, though)

Reply:
Thanks @fortmarek 

In that case two suggestions

- We can have a private method along the lines of `printAndFlushPendingLintWarnings` which would clear out the warnings after they get printed
- Alternatively have each of the methods that return lint issues return them and we'd manually aggregate them in the `generateWithGraph`

I suspect the first option might be a bit cleaner